src/components/network-status/NetworkStatus.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useState, useEffect } from 'react'
+
+import { NetworkStatusMessage } from './NetworkStatusMessage'
+
+export const NetworkStatus = () => {
```

Comment:
I wonder if we can tweak this function a bit and turn it into a hook that returns `networkStatus` for its increased flexibility in future use. This would be called `useNetworkStatus`, and `NetworkStatusMessage` would use it by:

`const networkStatus = useNetworkStatus()`

and in `HospitalRun.tsx`, `NetworkStatusMessage` would be used?

Reply:
Good idea, I'll do that :+1:

Reply:
Actually, I started to implement your suggestion and now I see that, if I use the useNetworkStatus hook inside the NetworkStatusMessage component, it will make it harder to test NetworkStatusMessage, because it won't be depending on any props anymore.
What do you think about using the useNetworkStatus hook in the HospitalRun component and sending the network status object into NetworkStatusMessage?
That way NetworkStatusMessage will still be easy to test.

Reply:
@alessioprestileo Ideally, we'll mock the value like this? https://www.richardkotze.com/coding/mocking-react-hooks-unit-testing-jest

Reply:
@kumikokashii Ok, I'll try that :+1: